src/track/track.cpp

```diff
@@ -1454,30 +1454,70 @@ CoverInfo Track::getCoverInfoWithLocation() const {
     return CoverInfo(m_record.getCoverInfo(), m_fileAccess.info().location());
 }
 
+namespace {
+
+bool isSafeToOverwriteMetadataOnExport(mixxx::TrackRecord::SourceSyncStatus sourceSyncStatus) {
+    switch (sourceSyncStatus) {
+    case mixxx::TrackRecord::SourceSyncStatus::Void:
+        // TODO: Decide on how to proceed if metadata has not been imported yet
+        // and add appropriate tests for this use case to LibraryFileSyncTest.
+        return false;
+    case mixxx::TrackRecord::SourceSyncStatus::Synchronized:
+        // If the metadata is still synchronized with the source exporting
+        // modified metadata from Mixxx is perfectly safe.
+        return true;
+    case mixxx::TrackRecord::SourceSyncStatus::Unknown:
+        // After the migration the sync status of all existing tracks in the
+        // library is unknown. In this case we assume that exporting metadata
+        // is desired independent of the last modification time of the file.
+        return true;
+    case mixxx::TrackRecord::SourceSyncStatus::Undefined:
+        // Trying to write into an inaccessible file is safe even though it
+        // might fail subsequently. We need to continue here for avoiding
+        // misleading or wrong error messages about the synchronization status!
+        return true;
+    default:
```

Comment:
Outdated is missing

Reply:
Adding a regression test for the missing enum variant revealed a conceptual problem that needs to be fixed.

Reply:
https://github.com/mixxxdj/mixxx/pull/4628/commits/ab2df29222468b71161acb28e2d74ef53cd7cda3

The comment in `Track::exportMetadata()` for the `Outdated` case explains why we have no choice other than overwriting the (maybe newer) metadata in the file and why this is a reasonable decision.

Reply:
It seems like metadata is not saved in the database if the file has disappeared (sync status = `Undefined`), regardless of the import/export synchronization settings. I had to disable the corresponding tests because this is a completely different issue.

Reply:
I have changed the behavior if export of metadata failed in https://github.com/mixxxdj/mixxx/pull/4628/commits/ffdddcb3f2e30bc5f28409fd9894822f16bb01dc. In this case the time stamp is now reset such that the sync status becomes `Unknown` when the file reappears in the future. That's the best we could do here right now.